### PR TITLE
Normalize spacing in flag help text.

### PR DESCRIPTION
**Pull request by petermattis**
[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/3981)

### CODE REVIEW

**Comment by BramGruneir** [comment]
LGTM

**Comment by bdarnell** [comment]
This is likely to break again due to people's editor configurations. It's standard to indent go code with tabs instead of spaces (and `gofmt` enforces this for code but not for string literals). It's difficult to robustly indicate an 8-space indent in a multi-line string literal, so maybe we should not include the indentation in the string literal but add it later when printing the help.

**Comment by petermattis** [comment]
That's a good point. Let me investigate removing the indentation entirely and re-adding it when outputting.

**Comment by RaduBerinde** [comment]
I assume they are being printed by PrintDefaults. To indent the first line, it uses 4 spaces followed by a tab:
https://github.com/golang/go/blob/f78a4c84ac8ed44aaf331989aa32e40081fd8f13/src/flag/flag.go#L449

Maybe we can define each usage as a []string { "line1", "line2" } and do strings.Join(flagUsage[f], "    \t")

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3981#-:-K9-Jftw-ccqPD3grTUI:2132264985)*

**Comment by petermattis** [comment]
The flags are being printed by [FlagSet.FlagUsages](https://github.com/spf13/pflag/blob/master/flag.go#L483) (from the `pflag` passage) which uses 6 spaces to indent the first line.

I've changed this PR to use `Wrap` and `Indent` functions from `github.com/kr/text`. PTAL.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3981#-:-K92XM3S_YwACE-FBA7J:-343967830)*

**Comment by tamird** [comment]
LGTM

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3981#-:-K92uLpbATqaIbsfKIcq:-533473425)*

**Comment by BramGruneir** [comment]
LGTM

**Comment by RaduBerinde** [comment]
Interesting, how does that happen? Shouldn't we be doing `import flag "github.com/ogier/pflag` instead of `import flag`?

The line you quoted (https://github.com/spf13/pflag/blob/master/flag.go#L483) indents the name of the flag 6 spaces, but the descriptions are indented below (https://github.com/spf13/pflag/blob/master/flag.go#L524) so that all descriptions align to the longest flag name..

In any case, if the output looks good, the updated change LGTM.

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/3981#-:-K92zgSv0MrlOa6wiDrf:-240702089)*

**Comment by petermattis** [comment]
The import of `flag` is so that we can map stdlib flags to `pflags`. See https://github.com/cockroachdb/cockroach/blob/master/cli/flags.go#L166-L169. I'm not sure what that indentation is supposed to be doing, but it doesn't seem to be doing anything. This is easy to verify experimentally. Or maybe that code isn't even being used to print the flag usage strings. Regardless, the output looks good after this change and it will be normalized to the correct indentation and wrap width independent of the last person who edited it.

File: cli/flags.go [hunk 1]
```diff
@@ -45,112 +46,113 @@ func (v pflagValue) IsBoolFlag() bool {
...
+Define the maximum number of results that will be retrieved.`),
+	"balance-mode": wrapText(`
+Determines the criteria used by nodes to make balanced allocation
+decisions. Valid options are "usage" (default) or "rangecount".`),
+	"password": wrapText(`
+The created user's password. If provided, disables prompting. Pass '-' to
+provide the password on standard input.`),
+}
+
+const wrapWidth = 79
```
Comment (bdarnell) [inline cli/flags.go]: This should be less than 79 to allow room for the indentation. The default values are also getting printed at the wrong indentation level.

Comment (petermattis) [inline cli/flags.go]: Will fix both issues in a follow-on PR.

**Pull request merged**